test/integration/src/test/java/io/pravega/test/integration/ReaderGroupNotificationTest.java

```
@@ -189,19 +213,128 @@ public void testSegmentNotifications() throws Exception {
+                                //Add segment event listener
+                                Listener<SegmentNotification> l1 = notification -> {
+                                    log.info("Number of Segments: {}, Number of Readers: {}", notification.getNumOfSegments(), notification.getNumOfReaders());
+                                    notificationResults.add(notification);
+                                };
+                                SegmentNotifier segmentNotifier = (SegmentNotifier) readerGroup.getSegmentNotifier(executor);
+                                segmentNotifier.registerListener(l1);
+                                segmentNotifier.pollNow();
+                                SegmentNotification initialSegmentNotification = notificationResults.poll();
+                                log.info("notification={}", initialSegmentNotification);
+                                readerGroup.resetReaderGroup(readerGroupConfig);
+                                readerGroup.readerOffline("readerId", null);
+                                long segmentSize = streamSegments.getSegments().size();
+                                log.info("segment_size={}", segmentSize);
+                                if (segmentSize > 3) {
+                                    log.info("success");
+                                    return;
+                                }
+                                throw new IllegalStateException("expected number of segments to be > 3");
+                            } catch (InterruptedException e) {
+                                log.error("Exception:", e);
+                                throw new IllegalStateException(e);
+                            }
+                        }), threadPoolExecutor)
+                .whenComplete((r, ex) -> {
+                        if (ex != null) {
+                            log.error("Failure: ", ex);
```

This is superfluous. Let the exception bubble up and the test will be automatically failed.

---

Yes Andre